src/Domain/Setting/SettingRepositoryInterface.ts

```diff
@@ -1,6 +1,10 @@
+import { CreateOrReplaceSettingDto } from './CreateOrReplaceSettingDto'
+import { CreateOrReplaceSettingStatus } from './CreateOrReplaceSettingStatus'
 import { Setting } from './Setting'
 
 export interface SettingRepositoryInterface {
   findOneByNameAndUserUuid(name: string, userUuid: string): Promise<Setting | undefined>
   findAllByUserUuid(userUuid: string): Promise<Setting[]>
+
+  createOrReplace(setting: CreateOrReplaceSettingDto): Promise<CreateOrReplaceSettingStatus>
```

Comment:
you can just add `save` ... some repos already have it in the interface and it's implement in the base class of typeorm repository

Reply:
hmm unless we need to indicate if it was updated/created - wdyt ?

Reply:
I am using typeorm's `Repository::save` in the implementation of this. Redeclaring a specialized version of it in this interface is possible, but feels pretty weird. And yes -- I specifically wanted to have a method for capturing the create or replace [semantics of PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT). This semantic is pretty common: [1](https://docs.microsoft.com/en-us/analysis-services/tmsl/createorreplace-command-tmsl?view=asallproducts-allversions), [2](https://docs.kuzzle.io/core/1/api/controllers/document/create-or-replace/), [3](https://javadoc.io/static/io.fabric8/kubernetes-client/1.4.29/io/fabric8/kubernetes/client/dsl/internal/CreateOrReplaceable.html) (random google hits for createorreplace).